frontend/src/concepts/areas/utils.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  DashboardConfigKind,
+  DashboardCommonConfig,
+  DataScienceClusterKindStatus,
+} from '~/k8sTypes';
+import { SupportedArea } from './types';
+import { SupportedAreasStateMap } from './const';
+
+type Output = { [flag: string]: boolean };
+const getFlags = (dashboardConfig: DashboardConfigKind): Output => {
+  const flags = dashboardConfig.spec.dashboardConfig;
+
+  return {
+    ...Object.keys(flags).reduce<Output>((flagState, key) => {
+      const value: boolean | string | undefined = flags[key as keyof DashboardCommonConfig];
+      if (typeof value === 'boolean') {
+        flagState[key] = key.startsWith('disable') ? !value : value;
```

Comment:
Do you want to just omit any flag that doesn't start with `disable` ?

Reply:
I see no value in it -- it'll just confuse the code more to mess with the state of everything. I just want a simple check state to compare against.

Reply:
OK, if you want to accept the entirety of the config.
I asked in slack about other properties that also get accepted as flags but aren't really flags: `modelMetricsNamespace` (which i believe is going away) and `enablement`.

Reply:
This is fair -- this could use with better typing. Right now it's isolated, maybe a follow up?